packages/ra-core/src/controller/input/useReferenceParams.ts

```diff
@@ -127,20 +127,28 @@ export const useReferenceParams = ({
     const filterValues = query.filter || emptyObject;
     const displayedFilterValues = query.displayedFilters || emptyObject;
 
-    const debouncedSetFilters = lodashDebounce((filter, displayedFilters) => {
-        changeParams({
-            type: SET_FILTER,
-            payload: {
-                filter: removeEmpty(filter),
-                displayedFilters,
-            },
-        });
-    }, debounce);
+    const debouncedSetFilters = useRef(
+        lodashDebounce((filter, displayedFilters) => {
+            changeParams({
+                type: SET_FILTER,
+                payload: {
+                    filter: removeEmpty(filter),
+                    displayedFilters,
+                },
+            });
+        }, debounce)
+    );
+    useEffect(() => {
+        return () => {
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+            debouncedSetFilters.current.cancel();
```

Comment:
As this effect runs every time the hoo runs, I'm afraid this will cancel all filters

Reply:
Shame on me.

Reply:
Now that makes sense: you clear the debounce on unmount. Got it!